waspc/data/Generator/templates/server/src/server.ts

```diff
@@ -21,12 +22,13 @@ const startServer = async () => {
   const port = normalizePort(config.port)
   app.set('port', port)
 
+  const server = http.createServer(app)
+
   {=# doesServerSetupFnExist =}
-  await {= serverSetupJsFnIdentifier =}()
+  const serverSetupFnContext: ServerSetupFnContext = { app, server }
```

Comment:
@shayneczyzewski Why are we exposing the `server` object? I'm not sure for which use case it's needed

Reply:
Makes sense, I though that `socket.io` didn't need the server 🤦‍♂️

Reply:
When I was doing this for an experiment, I did it as described here: https://discord.com/channels/686873244791210014/920312576133443634/1070052697572638870 -> so I passed `app` to setupFn, but I didn't move the server creation above it and then pass it.

I think it is cool that you did, that gives people more stuff to work with!

But it does change behavior a bit -> before, `setupFn` was allowing you to do stuff before `server` is created. Now, it allows you to set up stuff after server is created. I have no idea if we loose anything with that? Is there a use case where it might be important to do something before `server` is created from `app`? If we are not sure now, that is ok, we can just go with this and if we learn there is use case for it, we can add another function called `preSetupFn` or smth, that will be called just before the server.

One thing I thought was important was to add routes to `app` before the `server(app)` is called, but I guess I was wrong because I am sure you tested this and it works, even though you add stuff to `app` after the `server(app)` is called. Interesting. But does it work because it should, or by accident?

Maybe we can get some answers here: https://nodejs.org/api/http.html#class-httpserver .

Reply:
Good question @Martinsos - I don't think there is any technical reason why we have to keep it after, but you are right it does change the previous implied semantics a bit from user-level docs (which I massaged to say "before the server starts accepting requests"). But I don't think there is any actual technical impact in doing this swap. The custom route I setup in the `setupFn` still worked even when `createServer` happened before. I am not aware of any other potential impacts of this.

I believe `createServer` is just registering a request handler callback (of which an Express instance itself is). That callback should be using things internally that can be set either before or after that registration (i.e. it shouldn't even know it was registered to me). But I will do more testing and digging to be 100% sure (including doing a Socket.IO test too).

Reply:
It's not pretty (we have to specify the client/server URLs since different domains) but you can get Socket.IO up and running with access to `server`: https://gist.github.com/shayneczyzewski/165b4266d77a2a75454f605dd55b5db7